include/SFML/Graphics/Shader.hpp

```diff
@@ -58,8 +58,9 @@ class SFML_GRAPHICS_API Shader : GlResource, NonCopyable
     ////////////////////////////////////////////////////////////
     enum Type
     {
-        Vertex,  ///< Vertex shader
-        Fragment ///< Fragment (pixel) shader
+        Vertex =   1 << 0, ///< Vertex shader
+        Geometry = 1 << 1, ///< Geometry shader
+        Fragment = 1 << 2  ///< Fragment (pixel) shader
```

Comment:
I think this might essentially be a leftover. IIRC there used to be a compile function that would use an integer to determine what to compile. I could be wrong though, but wouldn't consider these values to be such a bad idea. They might be used later to store which shaders actually exist in the object for example.